src/Database/Redis/Connection.hs

```diff
@@ -276,19 +280,22 @@ refreshShardMap (Cluster.Connection nodeConns _ _ _ _) =
 
 refreshShardMapWithNodeConn :: [Cluster.NodeConnection] -> IO ShardMap
 refreshShardMapWithNodeConn [] = throwIO $ ClusterConnectError (Error "Couldn't refresh shardMap due to connection error")
-refreshShardMapWithNodeConn ((Cluster.NodeConnection ctx _ _) : xs) = do
+refreshShardMapWithNodeConn nodeConnsList = do
+    selectedIdx <- randomRIO (0, (length nodeConnsList) - 1)
+    let (Cluster.NodeConnection ctx _ _) = nodeConnsList !! selectedIdx
```

Comment:
we call refreshShardMap when one of the node is unavailable and with the random logic the problem is we might chose the same node again.

Reply:
once there's timeout in refreshShardMap in any nodeConnection we'll have to discard the connection (same reason as request timeout). So why not just try for 1 node and throwIO if failure.
we could hv just used `head nodeList ` but HM.elems gives list in ascending order of keys. so we'll end up refreshing using same node. 
Using random we are increasing the chances for a successful refreshShardMap. (not by much ofcourse)
Note - this is just a hot fix, it can we removed once  1 pool-1 node PR is merged

Reply:
Makes sense.